src/ScrollList.tsx

```diff
@@ -1,102 +1,61 @@
-  offset?: number;
-}
-
-export interface IScrollList extends IScrollListProps {}
-
-export const ScrollList: FunctionComponent<IScrollList> = props => {
-  const { list, offset, rowRenderer, onScroll, ...rest } = props;
-  const theme = useTheme();
-
-  const [{ clientHeight, scrollHeight, scrollTop }, setScrollEvent] = useState<IOnScroll>({});
-  const thumbSize = getThumbDimension({ scroll: scrollHeight, client: clientHeight });
 
-  const [isScrolling, setIsScrolling] = useState<null | number | NodeJS.Timer>(null);
-
-  const renderRow = ({ index, ...rowProps }: ListRowProps) => rowRenderer({ index, value: list[index], ...rowProps });
+export const FixedSizeList: FunctionComponent<IFixedSizeList> = props => {
+  const styles = scrollListStyles();
+  const { width, height, ...rest } = props;
 
   return (
-    <AutoSizer>
-      {({ height, width }) => (
-        <Box height={height} width={width} overflow="hidden">
-          <List
-            {...rest}
-            rowCount={list.length}
-            rowRenderer={renderRow}
-            onScroll={(e: IOnScroll) => {
-              setScrollEvent(e);
-
-              if (isScrolling !== null) {
-                clearTimeout(isScrolling as number);
-              }
-
-              setIsScrolling(
-                setTimeout(() => {
-                  setIsScrolling(null);
-                }, 1000)
-              );
+    <AutoSizer width={width} height={height}>
+      {({ width: listWidth, height: listHeight }) => (
+        <WindowFixedSizeList {...rest} css={styles} height={listHeight} width={listWidth} />
+      )}
+    </AutoSizer>
+  );
+};
 
-              if (onScroll) {
-                onScroll(e);
-              }
-            }}
-            scrollTop={offset}
-            height={height + 15} // add 15 to offset the native scrollbars
-            width={width + 15} // add 15 to offset the native scrollbars
-          />
+export const VariableSizeList: FunctionComponent<IVariableSizeList> = props => {
+  const styles = scrollListStyles();
+  const { width, height, ...rest } = props;
 
-          {/** scrollbar */}
-          <Box
-            {...verticalTrackStyle()}
-            height={`${thumbSize}px`}
-            width="6px"
-            cursor="grab"
-            borderRadius="10px"
-            opacity={isScrolling ? 1 : 0}
-            backgroundColor={theme.scrollbar.bg}
-            transform={`translateY(${getScrollTransform(clientHeight, scrollHeight, scrollTop, thumbSize)}px)`}
-            transition="opacity .1s"
-          />
-        </Box>
+  return (
+    <AutoSizer width={width} height={height}>
+      {({ width: listWidth, height: listHeight }) => (
+        <WindowVariableSizeList {...rest} css={styles} height={listHeight} width={listWidth} />
       )}
     </AutoSizer>
   );
 };
+
+export { IFixedSizeList, IVariableSizeList };
+
+export const scrollListStyles = () => {
+  const theme = useTheme();
+
+  return css`
+    &::-webkit-scrollbar {
```

Comment:
How consistent / inconsistent is this with the `ScrollBox` component approach / look and feel?

Reply:
The biggest inconsistency is a lack of autohiding. Scrollbar is always visible unlikes previously where it was visible only during scrolling.
Are we fine with it?

Chromium-based web browsers and Safari should have fairly equal UI. It can be always adjusted since we can treat scrollbar as it was a regular node and most of CSS rules can be applied.

Firefox and others - just limited styling possibilities.

Reply:
Ah interesting. Auto-hide is one of the primary reasons to do custom scrollbars - otherwise we end up with tons of gross scrollbars everywhere. VS code auto-hides - how do they do it if they're doing the css approach?

Reply:
They don't - I was wrong telling you they do, They have plain divs there and just update them as you scroll.
I think we can implement that too even when using react-window, as both `onScroll` and `scrollTo` are exposed.

Reply:
Looks like `react-window` might not support custom scrollbars: https://github.com/bvaughn/react-window/issues/110

I followed up to see if they were able to get it to work or not

Reply:
Was playing around that and found a solution to achieve autohiding. We can set transparent backgrond-color for scroll. Will finalize this approach.